fhir-path/src/main/java/org/linuxforhealth/fhir/path/evaluator/FHIRPathEvaluator.java

```diff
@@ -359,6 +359,39 @@ private Collection<FHIRPathNode> as(List<ExpressionContext> arguments) {
             }
             return result;
         }
+        
+        /**
+         * https://hl7.org/fhirpath/N1/#as-type-specifier
+         * https://hl7.org/fhirpath/N1/#astype-type-specifier
+         * This method returns the value of the left operand if it is of the type specified in the second operand.
+         * If the identifier cannot be resolved to a valid type identifier, the method will throw an error. 
+         * If there is more than one item in the input collection, the method will throw an error. 
+         * sOtherwise, this operator returns the empty collection
+         * @param arguments the parse tree
+         * @return the result of evaluation as a non-null, collection of FHIRPath nodes
+         */
+        private Collection<FHIRPathNode> asTypeEqual(List<ExpressionContext> arguments) {
+            if (arguments.size() != 1) {
+                throw unexpectedNumberOfArguments(arguments.size(), "as");
```

Comment:
```suggestion
                throw unexpectedNumberOfArguments(arguments.size(), "asTypeEqual");
```

Reply:
This is fixed as suggested. replaced "as" with "asTypeEqual".